src/main/java/redis/clients/jedis/ConnectionFactory.java

```diff
@@ -74,21 +66,9 @@ public PooledObject<Connection> makeObject() throws Exception {
     Connection jedis = null;
     try {
       jedis = new Connection(jedisSocketFactory, clientConfig);
-      jedis.connect();
       return new DefaultPooledObject<>(jedis);
     } catch (JedisException je) {
-      if (jedis != null) {
```

Comment:
If an exception is caught in new Connection, the following operations will be completed automatically and do not need to be executed again, see initializeFromClientConfig.